### ISSUE TITLE
About data augmentation

### ISSUE THREAD

**Issue by 4fee8fea** [likyoo/Multimodal-Remote-Sensing-Toolkit#2]
Hi, likyoo!

Thank you so much for such a solid code! I appreciate the repository very much.

When I am reading the function `mixture_noise` within `datasets.py` as follows:
````
    def mixture_noise(self, data, label, beta=1 / 25):
        alpha1, alpha2 = np.random.uniform(0.01, 1.0, size=2)
        noise = np.random.normal(loc=0.0, scale=1.0, size=data.shape)
        data2 = np.zeros_like(data)
        for idx, value in np.ndenumerate(label):
            if value not in self.ignored_labels:
                l_indices = np.nonzero(self.labels == value)[0]
                l_indice = np.random.choice(l_indices)
                assert self.labels[l_indice] == value
                x, y = self.indices[l_indice]
                data2[idx] = self.data[x, y]
        return (alpha1 * data + alpha2 * data2) / (alpha1 + alpha2) + beta * noise
````
**I found the following question:**
- `self.labels` is of the type `list`.
- `l_indices = np.nonzero(self.labels == value)[0]` may be not in accordance with the intent. 

However, although I fix `if self.mixture_augmentation and np.random.random() < 0.2` to `if self.mixture_augmentation` and run the script with `--mixture-augmentation`, the training procedure runs without exception, which confused me a lot.

**Another question I enountered is:**
I notice the `flip_augmentation` apply on `LiDAR` and `HSI` data both. But The `radiation_augmentation` and `mixture_augmentation` only apply on the `HSI` data, leaving `LiDAR` data alone. 
Could you please tell me the reason behind?

Best wishes!

**Comment by likyoo** [comment]
Hi, @4fee8fea thank you for your attention to our work.
This part of the data augmentation code is directly borrowed from DeepHyperX. Since it was not used in our experiments, it was not debugged..... And it seems that this part of the code does have some problems.
But we don't have enough time to perfect this work now...Could you debug this part of the code and pull the request? grateful :)

**Comment by 4fee8fea** [comment]
Thanks for your reply and trust! 

I just found the variable `value` is of type `np.uint8`.

Amazingly `list` data can be compared directly with `np.uint8` value and return an `numpy.ndarray` as result.

I also noticed the the paper mentioned `flip_augmentation` and `rotation_augmentation` only. 

> All possible flips and rotations multiple 90$^\angle$ are used to augment training data.

Maybe `radiation_augmentation` and `mixture_augmentation` are not used in the experiment.

Best wishes!
